cmd/vic-machine/create/create.go

```diff
@@ -547,28 +673,135 @@ func (c *Create) processInsecureRegistries() error {
 	return nil
 }
 
-func (c *Create) loadCertificate() (*certificate.Keypair, error) {
+func (c *Create) loadCertificates() ([]byte, *certificate.KeyPair, error) {
 	defer trace.End(trace.Begin(""))
 
-	var keypair *certificate.Keypair
+	c.envFile = fmt.Sprintf("%s.env", c.DisplayName)
+
+	// reads each of the files specified, assuming that they are PEM encoded certs,
+	// and constructs a byte array suitable for passing to CertPool.AppendCertsFromPEM
+	var certs []byte
+	for _, f := range c.clientCAs {
+		log.Infof("Loading CA from %s", f)
+		b, err := ioutil.ReadFile(f)
+		if err != nil {
+			err = errors.Errorf("Failed to load authority from file %s: %s", f, err)
+			return nil, nil, err
+		}
+
+		certs = append(certs, b...)
+	}
+
+	var keypair *certificate.KeyPair
 	if c.cert != "" && c.key != "" {
 		log.Infof("Loading certificate/key pair - private key in %s", c.key)
-		keypair = certificate.NewKeyPair(false, c.key, c.cert)
-	} else if !c.noTLS && c.DisplayName != "" {
-		c.key = fmt.Sprintf("./%s-key.pem", c.DisplayName)
-		c.cert = fmt.Sprintf("./%s-cert.pem", c.DisplayName)
-		log.Infof("Generating certificate/key pair - private key in %s", c.key)
-		keypair = certificate.NewKeyPair(true, c.key, c.cert)
+		keypair = certificate.NewKeyPair(c.cert, c.key, nil, nil)
+
+		if err := keypair.LoadCertificate(); err != nil {
+			log.Errorf("Failed to load certificate: %s", err)
+			return certs, nil, err
+		}
 	}
-	if keypair == nil {
-		log.Warnf("Configuring without TLS - to enable drop --no-tls or use --key/--cert parameters")
-		return nil, nil
+
+	return certs, keypair, nil
+}
+
+func (c *Create) generateCertificates(ca bool) ([]byte, *certificate.KeyPair, error) {
+	defer trace.End(trace.Begin(""))
+
+	var certs []byte
+	// generate the certs and keys with names conforming the default the docker client expects
+	// to avoid overwriting for a different vch, place this in a directory named for the vch
+	err := os.MkdirAll(fmt.Sprintf("./%s", c.DisplayName), 640)
```

Comment:
Should we use 600 for the perms, or do you want group read access?
And `Mkdir` instead of `MkdirAll`, so we don't overwrite if already exists?  Maybe use `MkdirAll` only if `-force` flag is given.